File: src/Systems/Go/singleton.go
package main

import (
	"fmt"
	"sync"
)

type processRegistry struct {
	count int
}

var (
	registryOnce sync.Once
	registryInst *processRegistry
)

func registry() *processRegistry {
	registryOnce.Do(func() { registryInst = &processRegistry{} })
	return registryInst
}

func main() {
	first := registry()
	second := registry()
	first.count++
	fmt.Printf("same=%t\n", first == second)
	fmt.Printf("count=%d\n", second.count)
}

File: src/Web/TypeScriptTS/singleton.ts
class ProcessRegistry {
  private static readonly shared = new ProcessRegistry();
  private countValue = 0;

  private constructor() {}

  static instance(): ProcessRegistry {
    return ProcessRegistry.shared;
  }

  increment(): void {
    this.countValue += 1;
  }

  count(): number {
    return this.countValue;
  }
}

const first = ProcessRegistry.instance();
const second = ProcessRegistry.instance();
first.increment();
console.log(`same=${first === second}`);
console.log(`count=${second.count()}`);
